cli/root.go

```diff
@@ -175,6 +175,15 @@ func versionCmd() *cobra.Command {
 	}
 }
 
+// setCliTest in TestMain to convey we are running in a test.
+func setCliTest() {
+	os.Setenv("CODER_TEST_CLI", "true")
+}
+
+func isCliTest() bool {
+	return os.Getenv("CODER_TEST_CLI") != ""
```

Comment:
Alternatively, you can look for a flag that gets registered by default on the test binary:
```suggestion
		return flag.Lookup("test.v") != nil
```